### ISSUE TITLE
Swear filter easily bypassed

### ISSUE THREAD

**Issue by nailuj29gaming** [jonas747/yagpdb#858]
The swear filter can be easily bypassed by putting a symbol before the word, it should detect `thisispooop` as a message with a swear, for example.

**Comment by Wolveric** [comment]
Have you considered this behaviour is intentional?

**Comment by jo3-l** [comment]
Accurate swear filters are not simplistic things to make, a consequence of ignoring word boundaries might be that the word embar**ass**ed would be filtered.

**Comment by nailuj29gaming** [comment]
Why would it be intentional? It defeats the point of the filter

**Comment by nailuj29gaming** [comment]
Also, why write your own, when swear filters already exist

**Comment by jo3-l** [comment]
I am genuinely interested by this claim. Please point me to a Golang implementation of a swear filter that is fast, memory-efficient, and covers more cases than the current one while not adding many false positives.

**Comment by Wolveric** [comment]
Yeah, no... Automod is design to **supplement** manual moderation, to leave a reduced workload. As Joe pointed out: Making highly accurate fillers is difficult, so it's designed to be _reasonable_, rather than attempting to be incredibly accurate.

**Comment by nailuj29gaming** [comment]
I don't know of one, but you could make it so that some words that will never appear in another word are banned if the message contains them, and others are found on boundaries.

**Comment by Wolveric** [comment]
Assuming Jonas would be comfortable with implementing and maintaining the infrastructure for the two variants of the system.

**Comment by nailuj29gaming** [comment]
Its not really two variants, more like different words have different regexes to handle them

**Comment by Wolveric** [comment]
Yeah, I think that makes it pretty evident you have no idea how this system works...  it doesn't, nor should it, use RegEx in the processing of terms. Plus, there's the case of the UI on the website, and considerations for databases... come back once you actually have a concrete idea for this.

**Comment by Wolveric** [comment]
That, or make a suggestion on [the support server.](https://discord.gg/4udtcA5)

**Comment by nailuj29gaming** [comment]
That was just an idea, there's probably a better way.

**Comment by jo3-l** [comment]
This seems to be a relatively reasonable request to me and can be implemented without adding excessive complexity / sacrificing memory usage / performance. For example, using `|` to denote a word boundary and checking upon a match (or during the matching itself). However, it does add a certain amount of complexity.
